lib/elixir/test/elixir/kernel/def_test.exs

```diff
@@ -0,0 +1,79 @@
+Code.require_file "../test_helper.exs", __DIR__
+
+defmodule Kernel.DefTest do
+  use ExUnit.Case, async: true
+
+  test "expects clauses for rescue, else, catch" do
+    assert_raise CompileError, ~r"expected -> clauses for :rescue in \"def\"", fn ->
```

Comment:
There is no need for a new test file, you can add one or two of those tests to `warning_test.exs`. Make sure to purge the module that was defined afterwards.

Reply:
Sure!

Reply:
I'm putting it on `error_test.exs` since it's not a warning and yes a compile error. OK?

Reply:
Yes, apologies.